app/routes/component-tree.js

```diff
@@ -8,21 +9,27 @@ export default TabRoute.extend({
   },
 
   model() {
-    return [];
+    return new Promise(resolve => {
+      this.port.one('view:viewTree', resolve);
+      this.port.send('view:getTree');
+    });
   },
```

Comment:
Now blocks until we get a tree, so we don't have to worry about the `viewTreeLoaded = false` state anymore